AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Support/Utils/PinpointRequestsRegistry.swift

```diff
@@ -70,7 +70,11 @@ private struct CustomPinpointHttpClientEngine: HttpClientEngine {
         let currentUserAgent = headers.value(for: userAgentHeader) ?? ""
         headers.update(name: userAgentHeader,
                        value: "\(currentUserAgent)\(userAgentSuffix)")
-        request.headers = headers
+        for header in headers.headers {
+            for value in header.value {
+                request.withHeader(name: header.name, value: value)
```

Comment:
Looking at how [this is implemented](https://github.com/smithy-lang/smithy-swift/pull/593/files#diff-d50c8a9f81bf3276120cb94e56812f016f1f0b657f9c28ca2e1adca7d4c5b460), it looks like we'll be adding all the original headers to the new `additionalHeaders` list, which will then be combined with the original headers resulting in unnecessary duplication.

I believe we can just do this instead:
```swift
request.withHeader(
    name: userAgentHeader, 
    value: "\(currentUserAgent)\(userAgentSuffix)"
)
```

But now I wonder, would we be sending two values for the `User-Agent` header then? 🤔. Would that still work?